lib/main.dart

```diff
@@ -67,20 +83,89 @@ class _SampleViewerAppState extends State<SampleViewerApp> {
         appBar: AppBar(
           title: const Text(title),
         ),
-        body: _ready
-            ? SampleListView(samples: _samples)
-            : const Center(child: Text('Loading samples...')),
+        body: Listener(
+          onPointerDown: (_) => FocusManager.instance.primaryFocus?.unfocus(),
```

Comment:
I'm not sure, but I think `onPointerDown` shouldn't be necessary. What is it trying to do? Is this to unfocus the search field and dismiss the keyboard? If it really is necessary, at least it should move down the widget tree and be closer to the `SampleListView`. Otherwise when you tap in the TextField, it immediately tries to unfocus it and then focuses it.

Reply:
Could it wrap the Expanded that wraps the SampleListView? Since it is part of the search mechanism, I'd prefer to keep it all together here in this file. But otherwise putting it inside SampleListView would be acceptable.

Reply:
Actually I just tried wrapping the SampleListView directly instead of the Expanded and this isn't throwing any exceptions!